src/Checkin.php

```diff
@@ -353,8 +353,8 @@ class="col-sm-2 control-label"><?= gettext('Adult Name(Optional)') ?></label>
                                  class="direct-chat-img initials-image">&nbsp
                             <a href="PersonView.php?PersonID=<?= $per->getPersonId() ?>"><?= $sPerson ?></a></td>
                         <td><?= date_format($per->getCheckinDate(), SystemConfig::getValue('sDateTimeFormat')) ?></td>
-                        <td><?= $sCheckinby ?></td>
-                        <td><?= date_format($per->getCheckoutDate(), SystemConfig::getValue('sDateTimeFormat')) ?></td>
+                        <td><?= $sCheckinby ?> </td>
+                        <td><?= !$per->getCheckoutDate() ? "" : $per->getCheckoutDate() ?? date_format($per->getCheckoutDate(), SystemConfig::getValue('sDateTimeFormat')) ?></td>
```

Comment:
I'm a little confused by this

If `$per->getCheckoutDate()` is false-y (meaning zero'd or null), return the empty string. Otherwise take the value of `$per->getCheckoutDate()` but if it's somehow null, use `date_format($per->getCheckoutDate(), SystemConfig::getValue('sDateTimeFormat'))`.

Can it ever use the get to the date_format call? What am I missing here?

Reply:
This i where the bug was, if the person is not checked out, then we crash

Reply:
I guess my question is, should this be simplified to:
```suggestion
                        <td><?= $per->getCheckoutDate() ? date_format($per->getCheckoutDate(), SystemConfig::getValue('sDateTimeFormat'))  : '' ?></td>
```

(phrased in words: if `$per->getCheckoutDate()` has a value, format it. otherwise, return the empty string)